### PR TITLE
14516 Fixed pending correct status + Cancel Payment style

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#14516

*Description of changes:*
- added check for new pending status
- added logic to display "processing..." while checking status
- reduced retry count from 10 to 5 to finish faster
- updated list item button hover colour
- added new filing status enum value
- added new  mixin method + unit test
- app version = 5.8.1

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/TodoList.vue [hunk 1]
```diff
@@ -130,9 +130,10 @@
                 </div>
 
                 <!-- pending filing - correction or alteration, or payment incomplete -->
-                <div v-else-if="isStatusPending(item)" class="todo-subtitle">
+                <div v-else-if="(isStatusPending(item) || isStatusPendingCorrection(item))" class="todo-subtitle">
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:133]: We didn't know about this status originally. Now we handle it correctly.

File: src/components/Dashboard/TodoList.vue [hunk 2]
```diff
@@ -130,9 +130,10 @@
...
                   <template v-if="isTypeCorrection(item) || isTypeAlteration(item)">
-                    <span>FILING PENDING</span>
+                    <span v-if="inProcessFiling === item.filingId">PROCESSING...</span>
+                    <span v-else>FILING PENDING</span>
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:136]: Like the other instances of filings where we check their status a few times a second apart, this one should say "processing..." while we check and "filing pending" after.

File: src/components/Dashboard/TodoList.vue [hunk 3]
```diff
@@ -728,11 +730,11 @@ export default class TodoList extends Vue {
...
   /**
    * Checks whether the subject filing is now Completed.
-   * Retries after 1 second for up to 10 iterations.
+   * Retries after 1 second for up to 5 iterations.
    */
   private checkIfCompleted (id: number, count: number): void {
-    // stop this cycle after 10 iterations
-    if (++count > 10) {
+    // stop this cycle after 5 iterations
+    if (++count > 5) {
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:737]: Ten iterations should be about 10 seconds... except when the API takes 5 seconds to return from each "GET filing" call, so this mitigates that.

File: src/components/Dashboard/TodoList.vue [hunk 4]
```diff
@@ -1756,7 +1758,9 @@ export default class TodoList extends Vue {
     color: $app-blue;
   }
 
-  #btn-draft-delete:hover {
+  #btn-draft-delete:hover,
+  #btn-delete-application:hover,
+  #btn-cancel-payment:hover {
     background-color: $gray1;
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:1764]: Hover styling was missing altogether from the Cancel Payment button.

The Delete Application button had default Vuetify styling, which is a very light gray. Now it's darker gray and consistent.

**Review by lewischenstudio — APPROVED** [review]

**Review by lambert-alex — COMMENTED** [review]
LGTM

**Pull request merged**